subsys/bluetooth/controller/crypto.c

```diff
@@ -46,32 +44,40 @@ int bt_rand(void *buf, size_t len)
 	return 0;
 }
 
-int bt_encrypt_le(const u8_t key[16], const u8_t plaintext[16], u8_t enc_data[16])
+int bt_encrypt_le(const u8_t key[16], const u8_t plaintext[16],
+		  u8_t enc_data[16])
 {
-	// @todo: support CONFIG_MULTITHREADING
-
 	LOG_HEXDUMP_DBG(key, 16, "key");
 	LOG_HEXDUMP_DBG(plaintext, 16, "plaintext");
 
-	uint32_t errcode = ble_controller_ecb_block_encrypt(key, plaintext, enc_data);
+	int32_t errcode = MULTITHREADING_LOCK_ACQUIRE();
+	if (!errcode) {
+		errcode = ble_controller_ecb_block_encrypt(key, plaintext, enc_data);
+		MULTITHREADING_LOCK_RELEASE();
+	}
 
 	LOG_HEXDUMP_DBG(enc_data, 16, "enc_data");
```

Comment:
The `LOG_HEXDUMP_DBG` should be moved inside `if (!errcode)`

Reply:
The same applies in other places

Reply:
Fixed.